File: src/main/java/dev/sky_lock/mocar/packet/OpenSignEditorPacket.java
package dev.sky_lock.mocar.packet;

import com.comphenix.protocol.PacketType;
import com.comphenix.protocol.wrappers.BlockPosition;

/**
 * @author sky_lock
 */

public class OpenSignEditorPacket extends ServerPacket {

    public OpenSignEditorPacket() {
        super(PacketType.Play.Server.OPEN_SIGN_EDITOR);
    }

    public void setBlockPosition(BlockPosition position) {
        getPacket().getBlockPositionModifier().write(0, position);
    }

}

File: src/main/kotlin/dev/sky_lock/pocketlifevehicle/packet/OpenSignEditorPacket.kt
package dev.sky_lock.pocketlifevehicle.packet

import com.comphenix.protocol.PacketType
import com.comphenix.protocol.wrappers.BlockPosition

/**
 * @author sky_lock
 */
class OpenSignEditorPacket : ServerPacket(PacketType.Play.Server.OPEN_SIGN_EDITOR) {
    fun setBlockPosition(position: BlockPosition) {
        packet.blockPositionModifier.write(0, position)
    }
}